### PR TITLE
Routes panel

### PR DESCRIPTION
**Pull request by andrej-griniuk**
Refs #457

### CODE REVIEW

File: src/Template/Element/routes_panel.ctp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <thead>
+    <tr>
+        <th><?= __d('debug_kit', 'Route name') ?></th>
+        <th><?= __d('debug_kit', 'URI template') ?></th>
+        <th><?= __d('debug_kit', 'Defaults') ?></th>
+    </tr>
+    </thead>
+    <tbody>
+    <?php foreach ($routes as $route): ?>
+        <tr<?php if ($matchedRoute === $route->template) echo ' class="highlighted"'; ?>>
```
Comment (dereuromark) [inline src/Template/Element/routes_panel.ctp]: Use <?= and ternary better for inliners.

File: src/Template/Element/routes_panel.ctp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        <th><?= __d('debug_kit', 'Route name') ?></th>
+        <th><?= __d('debug_kit', 'URI template') ?></th>
+        <th><?= __d('debug_kit', 'Defaults') ?></th>
+    </tr>
+    </thead>
+    <tbody>
+    <?php foreach ($routes as $route): ?>
+        <tr class="<?= ($matchedRoute === $route->template) ? 'highlighted' : '' ?>">
+            <td><?= \Cake\Utility\Hash::get($route->options, '_name', $route->getName()) ?></td>
+            <td><?= $route->template ?></td>
```
Comment (markstory) [inline src/Template/Element/routes_panel.ctp]: Should HTML escape this to be safe.

File: src/Template/Element/routes_panel.ctp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            <td><?= json_encode($route->defaults) ?></td>
```
Comment (markstory) [inline src/Template/Element/routes_panel.ctp]: Does `$this->Toolbar->makeNeatArray($route->defaults)` look OK here?

Comment (markstory) [inline src/Template/Element/routes_panel.ctp]: ok.

**Comment by lorenzo** [comment]
Would it be better to fetch the routes dynamically in the panel instead of storing it in the database for each request? Routes are more or less static data, but they consume space and also time to serialize

**Comment by andrej-griniuk** [comment]
Makes sense. Updated.

**Comment by markstory** [comment]
Thanks!

**Pull request merged**